feedback/extensions/filters.py

```diff
@@ -0,0 +1,175 @@
+        Args:
+            context (dict): the context for the instructor dashboard.
+            _ (str): instructor dashboard template name.
+        """
+        if not settings.FEATURES.get("ENABLE_FEEDBACK_INSTRUCTOR_VIEW", False):
+            return {
+                "context": context,
+            }
+
+        course = context["course"]
+        template = Template(self.resource_string(f"static/html/{TEMPLATE_CATEGORY}.html"))
+
+        request = get_current_request()
+
+        context.update(
+            {
+                "blocks": load_blocks(request, course),
+            }
+        )
+
+        html = template.render(Context(context))
+        frag = Fragment(html)
+        frag.add_css(self.resource_string(f"static/css/{TEMPLATE_CATEGORY}.css"))
+        frag.add_javascript(self.resource_string(f"static/js/src/{TEMPLATE_CATEGORY}.js"))
+
+        section_data = {
+            "fragment": frag,
+            "section_key": TEMPLATE_CATEGORY,
+            "section_display_name": "Course Feedback",
+            "course_id": str(course.id),
+            "template_path_prefix": TEMPLATE_ABSOLUTE_PATH,
+        }
+        context["sections"].append(section_data)
+
+        return {"context": context}
+
+    def resource_string(self, path):
+        """Handy helper for getting resources from our kit."""
+        data = pkg_resources.resource_string("feedback", path)
+        return data.decode("utf8")
+
+
+def load_blocks(request, course):
+    """
+    Load feedback blocks for a given course for all enrolled students.
+
+    Arguments:
+        request (HttpRequest): Django request object.
+        course (CourseLocator): Course locator object.
+    """
+    course_id = str(course.id)
+
+    feedback_blocks = modulestore().get_items(
+        course.id, qualifiers={"category": BLOCK_CATEGORY}
+    )
+
+    blocks = []
+    students = get_user_enrollments(course_id).values_list(
+        "user_id", "user__username"
+    )
+    for feedback_block in feedback_blocks:
+        block, _ = get_block_by_usage_id(
+            request,
+            str(course.id),
+            str(feedback_block.location),
+            disable_staff_debug_info=True,
+            course=course,
+        )
+        answers = load_xblock_answers(
+            request,
+            students,
+            str(course.location.course_key),
+            str(feedback_block.location),
+            course,
+        )
+
+        vote_aggregate = []
+        total_votes = 0
+        total_answers = 0
+
+        if not block.vote_aggregate:
+            block.vote_aggregate = [0] * len(block.get_prompt()["scale_text"])
+        for index, vote in enumerate(block.vote_aggregate):
+            vote_aggregate.append(
+                {
+                    "scale_text": block.get_prompt()["scale_text"][index],
+                    "count": vote,
+                }
+            )
+            total_answers += vote
+            total_votes += vote * (5-index)
```

Comment:
what's 5 - index? can we name that formula? so it's easier to understand

Reply:
It's because the likerts are not correctly sorted. (1, excellent, 5, poor). So we are inverting the values.